### DISCUSSION TITLE
How do I insert and delete a object Component of a Collection

### DISCUSSION THREAD

**Discussion by Marcosul** [strapi/strapi#6248] [Legacy Topics]
Hello

I'm really enjoying using the strapi, but I have a difficulty that I believe is basic

I created a component called Address and I would like to easily insert and edit Address objects using GraphQL. But I don't know how to do it.

I tried to use the updateUser mutation and pass a ComponentAddressInput object without the ID, but with each command, the objects are rewritten.

1) How do I insert a new object in the Address component? Do I use updateUser mutation or is there a component mutation? My code:

```
mutation setUser(
  $id: ID = "5ebf499ea13d2f52324903c1"
  $name: String = "Marco"
  $address: editComponentCoreAddressInput = {name: "myHome" info: "Number 123"}

  ) {

  updateUser(
    input: {
      where: { id: $id }
      data: { 
        name: $name
        addresses: [$address]

         }
    }
  ) {
    user {
      username,
      name
      addresses{
        id
        name
        info
      }
    }
  }
}
```

2) How do I delete an User Address Component object?

3) How do I customize a mutation with the resolver in the controller as well as the example of the documentation for the User since there is no folder controllers for the User?

**Comment by lauriejim** [comment]
Hello!

You will find all use cases there (REST API) but it's the same for GraphQL - [documentation](https://strapi.io/documentation/3.0.0-beta.x/content-api/api-endpoints.html#update-an-entry)

When you update a compo, you have to send all relations.

So in your case, in any  case you will have to send all your addresses, an array of address.
Important, you need to set the id of addresses in each existing addresses.

If you want to delete one, you will have to remove the one you want form your array.
If you want to add one, you will have to add it in the array (without ID key because this address doesn't exist)
If you want update and address, just update values of the address object (make sure the ID key is present.)

To customize the User - you will have to check the customisation concept, to be able to customize a plugin - [documentation](https://strapi.io/documentation/3.0.0-beta.x/concepts/customization.html)
